### PR TITLE
docs: Add the shortcut type of files

### PR DESCRIPTION
**Pull request by nono**
We want to use shortcuts/hyperlinks to make possible to have files thatare linked to a resource. The resource can be on the same cozy (internal link), on an external site (like wikipedia URL), or linked to a sharing.

### CODE REVIEW

File: docs/io.cozy.files.md [hunk 1]
```diff
@@ -168,6 +168,62 @@ The `io.cozy.files` doctype has [the standard `cozyMetadata`](https://docs.cozy.
 }
 ```
 
+### Shortcuts
+
+The shortcuts (or hyperlinks) are a special types of files. They are served
```
Comment (ptbrowne) [inline docs/io.cozy.files.md]:
```suggestion
Shortcuts (or hyperlinks) are a special type of files. They are served
```

File: docs/io.cozy.files.md [hunk 2]
```diff
@@ -168,6 +168,62 @@ The `io.cozy.files` doctype has [the standard `cozyMetadata`](https://docs.cozy.
...
+with the `application/shortcut` mime-type, they have a `.url` extension, and
+respect the [URL file
+format](http://www.lyberty.com/encyc/articles/tech/dot_url_format_-_an_unofficial_guide.html).
+
+They can have some `metadata`:
```
Comment (ptbrowne) [inline docs/io.cozy.files.md]:
```suggestion
They can have `metadata`:
```

File: docs/io.cozy.files.md [hunk 3]
```diff
@@ -168,6 +168,62 @@ The `io.cozy.files` doctype has [the standard `cozyMetadata`](https://docs.cozy.
...
+
+- `sharing-status` will be set if the link can be used to open a sharing. 2
+  values are possible: `new` and `seen`, depending on the fact that the link
```
Comment (ptbrowne) [inline docs/io.cozy.files.md]:
```suggestion
  values are possible: `new` and `seen`, depending on whether the link
  has been opened.
```

File: docs/io.cozy.files.md [hunk 4]
```diff
@@ -168,6 +168,63 @@ The `io.cozy.files` doctype has [the standard `cozyMetadata`](https://docs.cozy.
...
+with the `application/shortcut` mime-type, they have a `.url` extension, and
+respect the [URL file
+format](http://www.lyberty.com/encyc/articles/tech/dot_url_format_-_an_unofficial_guide.html).
+
+They can have `metadata`:
+
+- `sharing-status` will be set if the link can be used to open a sharing. 2
+  values are possible: `new` and `seen`, depending on whether the link
+  has been opened.
+  has been opened or not.
```
Comment (ptbrowne) [inline docs/io.cozy.files.md]:
```suggestion
```

File: docs/io.cozy.files.md [hunk 5]
```diff
@@ -168,6 +168,63 @@ The `io.cozy.files` doctype has [the standard `cozyMetadata`](https://docs.cozy.
 }
 ```
 
+### Shortcuts
+
+Shortcuts (or hyperlinks) are a special type of files. They are served
+with the `application/shortcut` mime-type, they have a `.url` extension, and
```
Comment (edas) [inline docs/io.cozy.files.md]: Je suis étonné du type mime. Le plus courant m'a l'air application/internet-shortcut. J'en trouve d'autres, mais aucun application/shortcut

File: docs/io.cozy.files.md [hunk 6]
```diff
@@ -168,6 +168,63 @@ The `io.cozy.files` doctype has [the standard `cozyMetadata`](https://docs.cozy.
...
+respect the [URL file
+format](http://www.lyberty.com/encyc/articles/tech/dot_url_format_-_an_unofficial_guide.html).
+
+They can have `metadata`:
+
+- `sharing-status` will be set if the link can be used to open a sharing. 2
```
Comment (edas) [inline docs/io.cozy.files.md]: How do you know ? It could have been opened in desktop, and then not catched

File: docs/io.cozy.files.md [hunk 7]
```diff
@@ -168,6 +168,63 @@ The `io.cozy.files` doctype has [the standard `cozyMetadata`](https://docs.cozy.
...
+  values are possible: `new` and `seen`, depending on whether the link
+  has been opened.
+  has been opened or not.
+- `target-cozy` is set to the cozy instance when relevant (mostly sharing
```
Comment (edas) [inline docs/io.cozy.files.md]: should we have a root `target: { … }` for all this instead of `metadata: { target-* }` properties ?

Comment (edas) [reply docs/io.cozy.files.md]: If so, should we try to mimic an actual file in this `target` ? Like : 

```
{
  "_id": "629fb233be550a21174ac8e19f0043af",
  "_rev": "1-61c7804bdb4f9f8dae5a363cb9a30dd8",
  "type": "file",
  "name": "sunset.jpg.url",
  "dir_id": "629fb233be550a21174ac8e19f003e4a",
  "trashed": false,
  "md5sum": "vfEMDpJShs8QeIlsDmw9VA==",
  "created_at": "2020-02-10T20:38:04Z",
  "updated_at": "2020-02-10T20:38:04Z",
  "tags": [],
  "target": {
    "mime": "image/jpg",
    "_type": "io.cozy.files",
    "cozyMetadata": {
      "createdOn": "https://alice.cozy.example/",
     }
  },
  "metadata": {
    "sharing-status": "new",
  },
  "size": 62,
  "executable": false,
  "class": "shortcut",
  "mime": "application/shortcut",
  "cozyMetadata": {
    "doctypeVersion": 1,
    "metadataVersion": 1,
    "createdAt": "2020-02-10T20:38:04Z",
    "createdOn": "https://bob.cozy.example/",
    "updatedAt": "2020-02-10T20:38:04Z",
    "uploadedAt": "2020-02-10T20:38:04Z",
    "uploadedOn": "https://bob.cozy.example/"
  }
}
```

File: docs/io.cozy.files.md (hunk 5 above)
Comment (nono) [reply docs/io.cozy.files.md]: C'est ce que j'avais noté quand on en discuté ensemble : https://trello.com/c/bPmXyElo/371-%F0%9F%93%81-drive-%E2%98%81%EF%B8%8F-stack-%F0%9F%94%97-hyperlien-jh-2-3j#comment-5e330f048191660578b302d1, mais ça me va d'utiliser `application/internet-shortcut` (possible que je me sois trompé en prennant les notes).

File: docs/io.cozy.files.md (hunk 6 above)
Comment (nono) [reply docs/io.cozy.files.md]: I'm not decided, but an option would be that the sharing link goes to the recipient cozy, and then redirects to the sharer cozy instance. The product team would like to show this information in Drive Web, so I'm trying to do that.

File: docs/io.cozy.files.md (hunk 7 above)
Comment (nono) [reply docs/io.cozy.files.md]: We could have: 

```json
"metadata": {
  "target": {
    "mime": "image/jpg",
    "_type": "io.cozy.files",
    "cozyMetadata": {
      "createdOn": "https://alice.cozy.example/",
     }
  },
  "sharing-status": "new"
}
```

But I would prefer to keep those attributes in the metadata. We need to allow web apps to give those fields, and metadata is good place for that. For example, for a link to `https://alice-calendar.cozy.example/#/123456`, the stack won't know how to fill the mime-type (is it an event or a calendar?).

Comment (nono) [reply docs/io.cozy.files.md]: And I don't think that `cozyMetadata.createdOn` is a good way to indicate the target cozy. Technically, the destination file can have been created somewhere else (like on dave cozy instance). It's just that Bob have a link that goes to Alice instance, but it doesn't say anything about the past of this target.

Comment (edas) [reply docs/io.cozy.files.md]: Ok for me in metadata if you prefer. Having a structured `target` will let us fill it with  more later, and still reuse some functions we already have `target`having the same structure as a standard file)

@Crash-- what do you think?

File: docs/io.cozy.files.md (hunk 5 above)
Comment (edas) [reply docs/io.cozy.files.md]: Je pense qu'on a du avoir une perte de communication quelque part. Pas très grave, la PR permet  de corriger

File: docs/io.cozy.files.md (hunk 7 above)
Comment (edas) [reply docs/io.cozy.files.md]: > And I don't think that `cozyMetadata.createdOn` is a good way to indicate the target cozy

:+1:
`ownedBy` ?

Comment (Crash--) [reply docs/io.cozy.files.md]: I like the idea of having something as close as possible to regular file. 

Also, having a `scope` target in the metadata object is more readable & understandable I think. Also easier to delete if needed

**Pull request merged**